crates/router/src/services.rs

```diff
@@ -46,6 +58,20 @@ impl Store {
         format!("{{{}}}_{}", shard_key, self.config.drainer_stream_name,)
     }
 
+    pub fn redis_conn(
+        &self,
+    ) -> errors::CustomResult<Arc<redis_interface::RedisConnectionPool>, RedisError> {
+        if self
+            .redis_conn
+            .is_redis_available
+            .load(atomic::Ordering::SeqCst)
+        {
+            Ok(self.redis_conn.clone())
```

Comment:
We can removing this clone() right?

Reply:
Since we take `&self` in this function and redis_conn doesn't implement copy. `redis_conn` will be moved out of self if we don't use clone here which will throw error

Reply:
And since this is an `Arc` clone is very less expensive